### DISCUSSION TITLE
Error predicting with n_lags>0

### DISCUSSION THREAD

**Discussion by Arthou** [ourownstory/neural_prophet#731] [Q&A - get help using NeuralProphet] [answered]
I am having a problem making my prediction work with ` n_lags`  in NeuralProphet. Below is a snippet of the dataframe I am using. It contains an hourly data of a company's energy consumption. 

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/RELAIS~1/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/RELAIS~1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

</head>

<body link=blue vlink=purple>

ds | y
-- | --
2020-09-01   01:00:00 | 430,5
2020-09-01   02:00:00 | 424,2
2020-09-01   03:00:00 | 429,24
... | ...
2022-02-31   22:00:00 | 415,8
2022-02-31   23:00:00 | 413,28
2022-02-31   00:00:00 | 408,66

The code below is the code I am using to predict the energy consumption of the next month, March of 2022. The dataframe presented above is saved to the `data` variable.

```
m = NeuralProphet(growth='on',
    n_lags=3*24,
    learning_rate=0.01)

m.add_country_holidays(country_name='BR')
metrics = m.fit(df=data, freq = 'H')
future = m.make_future_dataframe(df=data, periods=31*24, n_historic_predictions = True)
forecast = m.predict(future)
fig = m.plot(forecast)
plt.show()
```

The code works as expected when `n_lags=3*24` is commented, generating the graphic below:

![WithoutN_lags](https://user-images.githubusercontent.com/4604311/176537888-4e4f2bef-e62f-44c5-ae00-5dee2e354949.png)

But as soon as that line is present again, the code only predicts one additional period after the dataframe, independently of what is put in the `periods` argument in `m.make_future_dataframe`.

![WithN_lags](https://user-images.githubusercontent.com/4604311/176538238-46905cf9-22b7-4143-a4e5-d69c2d41b44f.png)

I don't know if this is intended behavior when the regressor is added, but I would appreciate any help to solve this issue.
</body>

</html>

**Comment by AchilleSoulie** [comment, marked as answer]
This is the normal behavior.
Using the network with the default value for  n_forecast = 1 will only allow you make one-step forecast into the future.
If you want to make 3*24 predictions, then you should specify n_forecast = 3*24 (this way the network learns to make prevision n_forecast step ahead in the future).
More on this topic here: https://neuralprophet.com/html/autoregression_yosemite_temps.html#Multi-step-forecast

**Reply by ourownstory** [reply]
Thank you @AchilleSoulie for the answer!
I would just add to this that it is best to forecast as far ahead as actually needed at a time, e.g. 24h, as the model learns an explicit relationship to each forecast step.
